Add Energy Tracker integration documentation (#42031)
home-assistant/home-assistant.github.io: source/_integrations/energy_tracker.markdown

---
title: Energy Tracker
description: Automatically send meter readings from Home Assistant to your Energy Tracker account.
ha_category:
  - Energy
ha_release: 2025.12
ha_iot_class: Cloud Push
ha_config_flow: true
ha_codeowners:
  - '@energy-tracker'
ha_domain: energy_tracker
ha_platforms:
  - diagnostics
ha_integration_type: service
---

The **Energy Tracker** {% term integration %} allows you to automatically send meter readings from Home Assistant sensors to your [Energy Tracker](https://www.energy-tracker.best-ios-apps.de) account.

Energy Tracker is an energy monitoring service with over 100,000 users that helps you track and analyze your energy consumption.

> 

Author: done

## Prerequisites

Before configuring the integration, you need:

1. An [Energy Tracker account](https://www.energy-tracker.best-ios-apps.de)
2. A **Personal Access Token** from your Energy Tracker account:
   1. Log in to your Energy Tracker account

>    2. Navigate to **API** → **Access Tokens**

Reviewer: ```suggestion
   2. Navigate to **API** > **Access Tokens**.
```
- use > in breadcrumbs.
- end sentences with a period.

Author: done

   3. Create a new Personal Access Token
   4. Copy and save the token securely

{% include integrations/config_flow.md %}

{% configuration_basic %}
Account name:
    description: "A descriptive name for your Energy Tracker account."
Personal Access Token:
    description: "Your Personal Access Token from Energy Tracker. You can create one in your Energy Tracker account under **API** → **Access Tokens**."
{% endconfiguration_basic %}

You can add multiple Energy Tracker accounts by repeating the setup process.

## Getting your device ID

To send meter readings, you need your **Standard Measuring Device ID**:

### Via Energy Tracker web interface

1. Log into your Energy Tracker account
2. Go to your device details
3. Copy the Standard Measuring Device ID
4. Remove the `std-` prefix from the ID
5. The final ID should be in UUID format: `xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`

### Via API (recommended)

1. Log into your Energy Tracker account
2. Navigate to **API** → **Documentation**
3. Use the API endpoint to retrieve your devices
4. The IDs returned are already in the correct format

## Actions

### Action: `energy_tracker.send_meter_reading`

> The `energy_tracker.send_meter_reading` action is used to send a meter reading from a Home Assistant sensor to Energy Tracker.

Author: done

- **Data attribute**: `entry_id`
  - **Description**: The Energy Tracker account entry ID to use for sending the meter reading.
  - **Optional**: No
- **Data attribute**: `device_id`
  - **Description**: The standard measuring device ID in UUID format from your Energy Tracker account.
  - **Optional**: No
- **Data attribute**: `source_entity_id`
  - **Description**: The entity ID of the Home Assistant sensor that provides the meter reading.
  - **Optional**: No
- **Data attribute**: `allow_rounding`
  - **Description**: Whether to round the value to match the meter precision. Defaults to `true`.
  - **Optional**: Yes

### Supported entity types

The action accepts meter readings from:

- Sensors (`sensor.*`)
- Input numbers (`input_number.*`)
- Number entities (`number.*`)

The entity state must be numeric and not `unavailable` or `unknown`.

## Automation examples

### Daily electricity reading

Send your electricity meter reading daily at 23:55:

{% raw %}
```yaml
- alias: "Send daily electricity reading"
  triggers:
    - trigger: time
      at: "23:55:00"
  conditions:
    - "{{ states('sensor.electricity_meter') != 'unavailable' }}"

>     - "{{ states('sensor.electricity_meter') != 'unknown' }}"

Author: Combining them into a single condition would exceed the 80-character line limit specified in the style guide for YAML code blocks.

---
title: Energy Tracker
description: Automatically send meter readings from Home Assistant to your Energy Tracker account.
ha_category:
  - Energy

> ha_release: 2025.12

Reviewer: ```suggestion
ha_release: 2026.1
```

Author: done

ha_iot_class: Cloud Push
ha_config_flow: true
ha_codeowners:
  - '@energy-tracker'
ha_domain: energy_tracker
ha_platforms:
  - diagnostics
ha_integration_type: service
---

The **Energy Tracker** {% term integration %} makes it easy for you to monitor and analyze your home's energy usage by automatically sending meter readings from your Home Assistant sensors to your [Energy Tracker](https://www.energy-tracker.best-ios-apps.de) account.

> Energy Tracker is a popular energy monitoring service, trusted by over 100,000 people, that helps you track your electricity, gas, and water consumption over time. With clear charts and insights, you can better understand your usage patterns and find ways to save energy.

Reviewer: ```suggestion
Energy Tracker is a popular energy monitoring service, trusted by over 100,000 people. It helps you track your electricity, gas, and water consumption over time. With clear charts and insights, you can better understand your usage patterns and find ways to save energy.
```

Author: done

## Prerequisites

Before configuring the integration, you need:

1. An [Energy Tracker account](https://www.energy-tracker.best-ios-apps.de)
2. A **Personal Access Token** from your Energy Tracker account:
   1. Log in to your Energy Tracker account.
   2. Navigate to **API** > **Access Tokens**.
   3. Create a new Personal Access Token.
   4. Copy and save the token securely.

{% include integrations/config_flow.md %}

{% configuration_basic %}
Account name:
    description: "A descriptive name for your Energy Tracker account."
Personal Access Token:
    description: "Your Personal Access Token from Energy Tracker. You can create one in your Energy Tracker account under **API** > **Access Tokens**."
{% endconfiguration_basic %}

You can add multiple Energy Tracker accounts by repeating the setup process.

## Getting your device ID

To send meter readings, you need your **Standard Measuring Device ID**:

### Via Energy Tracker web interface

1. Log into your Energy Tracker account.
2. Go to your device details.

> 3. Copy the Standard Measuring Device ID.

Reviewer: ```suggestion
3. Copy the **Standard Measuring Device ID**.
```
bold, if this is a UI string. If it isn't, I wonder whether the heading style capitalization is needed there.

Author: its the Device Identifier.
<img width="1140" height="196" alt="Image" src="https://github.com/user-attachments/assets/e721b31c-9909-4c92-95e7-03520d2c949c" />

4. Remove the `std-` prefix from the ID.
5. The final ID should be in UUID format: `xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`.

### Via API (recommended)

1. Log into your Energy Tracker account.
2. Navigate to **API** > **Documentation**.
3. Use the API endpoint to retrieve your devices.
4. The IDs returned are already in the correct format.

## Actions

> ### Action: `energy_tracker.send_meter_reading`

Reviewer: ```suggestion
### Action: Send meter reading
```

Author: done

> Sends a meter reading from a Home Assistant sensor to Energy Tracker.

Reviewer: ```suggestion
The `energy_tracker.send_meter_reading` action sends a meter reading from a Home Assistant sensor to Energy Tracker.
```
We are trying to move towards using the non-YAML name in the title, and use the YAML string in the intro paragraph.
https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L139

Author: done

## Automation examples

### Daily electricity reading

Send your electricity meter reading daily at 23:55:

{% raw %}
```yaml
- alias: "Send daily electricity reading"
  triggers:
    - trigger: time
      at: "23:55:00"
  conditions:
    - "{{ states('sensor.electricity_meter') != 'unavailable' }}"
    - "{{ states('sensor.electricity_meter') != 'unknown' }}"
  actions:
    - action: energy_tracker.send_meter_reading
      data:
        entry_id: "01234567890abcdef01234567890abcd"
        device_id: "deadbeef-dead-beef-dead-beefdeadbeef"
        source_entity_id: sensor.electricity_meter
```
{% endraw %}

### Gas meter on state change

Send gas meter reading when the sensor value changes:

{% raw %}
```yaml
- alias: "Send gas reading on change"
  triggers:
    - trigger: state
      entity_id: sensor.gas_meter
  conditions:
    - "{{ trigger.to_state.state not in ['unavailable', 'unknown'] }}"
  actions:
    - action: energy_tracker.send_meter_reading
      data:
        entry_id: "01234567890abcdef01234567890abcd"
        device_id: "12345678-1234-5678-1234-567812345678"
        source_entity_id: sensor.gas_meter
```
{% endraw %}

## Data updates

> Meter readings are sent on-demand via action calls. The integration does not poll Energy Tracker for data.

Reviewer: ```suggestion
Meter readings are sent on-demand via action calls. The integration does not {% term polling polls %} Energy Tracker for data.
```

Author: done

## Troubleshooting

### Authentication errors

If you see authentication errors after setup:

1. Go to {% my integrations title="**Settings** > **Devices & Services**" %}.

> 2. Find your Energy Tracker integration.

Reviewer: ```suggestion
2. Find your **Energy Tracker** integration.
```

Author: done

> 3. Click the {% icon "mdi:dots-vertical" %} menu and select **Reconfigure**.

Reviewer: ```suggestion
3. Select the {% icon "mdi:dots-vertical" %} menu and select **Reconfigure**.
```
We prefer [select over click](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/c/click)

Author: done
